### ISSUE TITLE
Loading the same rom twice with single screen mode unchecked results in a memory leak

### ISSUE THREAD

**Issue by GlacieAwn** [Lime3DS/Lime-3DS-Emulator#173]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

Canary 2111

### Description of Issue

loading the same rom more than once with single screen mode unchecked in the view tab results in a memory leak. this seems to trigger with both my own homebrew code, and commercial dumps from my own 3DS using gm9. I have attatched a screenshot of the CLI output.

I did try to fork the repo and fix it myself, however, I couldn't locate the issue. 
![image](https://github.com/Lime3DS/Lime3DS/assets/168385216/3dcb4385-679e-43b5-b77a-4d98de34022a)

### Expected Behavior

pretty self explanatory. it seemed to only happen when loading .3dsx files, not when loading installed .cia files. 

### Reproduction Steps

- open .3dsx rom with single screen mode unchecked
- close it
- open it again

### Log File

[lime3ds_log.txt](https://github.com/Lime3DS/Lime3DS/files/15239888/lime3ds_log.txt)

### System Configuration

CPU: Intel core i3 10100 4.300GHz
GPU/Driver: Intel CometLake-S GT2 [UHD Graphics 630] | Mesa-Graphics
RAM: 8GB
OS: Linux Mint 21.3 x86_64

**Comment by rtiangha** [comment]
Just curious, but does the issue still occur with this build (download links at the bottom)?

https://github.com/rtiangha/Lime3DS/actions/runs/8992289841

If not, then one of the final Canary changes that didn't get a chance to migrate to Nightly before the shutdown might have fixed it. If that's the case, I can issue a PR. If it doesn't fix it, then no ideas here, lol.

**Comment by GlacieAwn** [comment]
Hi, thanks for getting back to me so soon! I had to build from source myself using the build you sent, and I can confirm that it does not fix the issue. 

this is so weird. I have no idea what could be causing it. as stated before, I did fork and dive into the code to see if I could fix it myself, but I couldn't figure it out. if needed, I will periodically come back to it to see if I can figure it out. but for now, I'm going to close this issue. thanks for the quick response again!
